packages/css/src/utils/__tests__/class-name-generator.test.ts

```diff
@@ -0,0 +1,25 @@
+import { ClassNameGenerator } from '../class-name-generator';
+
+describe('ClassNameGenerator', () => {
+  it('should generate class names with minimal length', () => {
+    const generator = new ClassNameGenerator();
+    Array.from(Array(27).keys()).forEach(() => {
+      const className = generator.generateClassName('foo');
+      expect(className.length).toBe(1);
+    });
+  });
+
+  it('should skip reservedClassNames', () => {
+    const generator = new ClassNameGenerator({ reservedClassNames: ['a', 'b', 'c'] });
+    const className = generator.generateClassName('foo');
+    expect(className).toBe('d');
+  });
+
+  it('should generate class names starting with a number', () => {
```

Comment:
I guess this is safe because the class will be prefixed with `_`

Reply:
Ah, good catch. The test case should be `should NOT generate class names starting with a number`. 🤦‍♂️
I still haven't made the mind of whether prefix with `_`. Prefixing with `_` guarantee the uniqueness but reduce the perf improvement. 
I think I will provide a option and let the library consumer to decide.

Reply:
+1 on making it optional, sounds a perfect solution for different consumers :)